encoders/numeric/CompressionVaeEncoder/__init__.py

```diff
@@ -16,68 +18,63 @@ class CompressionVaeEncoder(BaseNumericEncoder, BaseTFEncoder):
-                 feature_normalization=False, tb_logging=False,
+    def __init__(self, model_path='temp',
                  *args, **kwargs):
         """
-        :param model_path: specifies the path to the model, and also acts as the model name.
-        :param X : array, shape (n_samples, n_features)
-            Training data for the VAE.
-            Alternatively, X can be the path to a root-directory containing npy files (potentially nested), each
-            representing a single feature vector. This allows for handling of datasets that are too large to fit
-            in memory.
-            Can be None (default) only if a model with this name has previously been trained. Otherwise None will
-            raise an exception.
-        :param train_valid_split: controls the random splitting into train and test data. Here 99% of X is used
-        for training, and only 1% is reserved for validation.
-        :param dim_latent: specifies the dimensionality of the latent space.
-        :param iaf_flow_length: controls how many IAF(Inverse Autoregressive Flow) flow layers the model has.
-        :param cells_encoder: determines the number, as well as size of the encoders fully connected layers.
-        :param initializer: controls how the model weights are initialized, with options being `orthogonal` (default),
-        `truncated_normal`, and `lecun_normal`
-        :param batch_size: determine the batch sizes used for training.
-        :param batch_size_test: determine the batch sizes used for testing.
-        :param feature_normalization: tells CVAE whether it should internally apply feature normalization
-        (zero mean, unit variance, based on the training data) or not. If True, the normalisation factors are stored
-        with the model and get applied to any future data.
-        :param tb_logging: determines whether the model writes summaries for TensorBoard during the training process.
+        :param model_path: specifies the path to the pretrained model
+
         """
         super().__init__(*args, **kwargs)
-        self.X = X
-        self.train_valid_split = train_valid_split
-        self.output_dim = output_dim
-        self.iaf_flow_length = iaf_flow_length
-        self.cells_encoder = cells_encoder
-        self.initializer = initializer
-        self.batch_size = batch_size
-        self.batch_size_test = batch_size_test
         self.model_path = model_path
-        self.feature_normalization = feature_normalization
-        self.tb_logging = tb_logging
 
     def post_init(self):
-        super().post_init()
-        self.to_device()
+        from cvae import cvae
+        import cvae.lib.model_iaf as model
+        import tensorflow as tf
+
+        params_path = os.path.join(self.model_path, 'params.json')
+
+        config = tf.ConfigProto(log_device_placement=False)
+        config.gpu_options.allow_growth = True
+
+        with tf.Graph().as_default():
+            if os.path.exists(params_path):
```

Comment:
Raise exception otherwise

Reply:
I raised the exception, but now we are always getting that exception because we don't have an existent pre-trained model. How did you manage it for the CustomImageTorchEncoder since the CI calls it with an empty model_path?

Reply:
I did pickled a modem as a preprocess of the test and provide that model path in the constructor of the encoder. I think such a logic can also be applied here?

Reply:
Yes, the problem is not the unit tests where I did what you said and they succeed, but when the container runs again and no model_path is provided.